syncweek-react-aad/react-aad
# Generate type definition files as part of build

## Purpose
* Generate type definition files as part of build process so consumers of the npm package can make use of type information.

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Execute build process, verify the `.d.ts` files are output in the `dist` folder
* Reference the dist folder as a module in a React/TypeScript file (much like the sample app does) and import the `LoginType` enum. Verify that the enum values can be discovered with intelisense. 

## Other Information
* I initially thought to add the `--declaration` flag to the npm `build` script to generate the definition files. While there, I cleaned up some of the other scripts which were using the method of referencing dependency executables which was popular before npm started adding executables to the PATH variable. Ultimately I decided the best place for the `declaration` flag was in the typescript config, but I kept the script fixes in for cleanliness.
